action.yml

```diff
@@ -0,0 +1,213 @@
+            org.opencontainers.image.description=${{ github.event.repository.description }}
+            org.opencontainers.image.vendor="Sentry"
+            org.opencontainers.image.revision=${{ github.pull_request.head.sha || github.sha }}
+            org.opencontainers.image.version=${{ github.pull_request.head.sha || github.sha }}
+            org.opencontainers.image.source=${{ github.server_url }}/${{ github.repository }}
+            org.opencontainers.image.url=${{ github.server_url }}/${{ github.repository }}
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@b5ca514318bd6ebac0fb2aedd5d36ec1b5c232a2 # v3.10.0
+
+    - name: Login to GHCR
+      if: steps.setup.outputs.publish_to_ghcr == 'true'
+      uses: docker/login-action@74a5d142397b4f367a81961eba4e8cd7edddf772 # v3
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ github.token }}
+
+
+    - name: Setup Google Cloud authentication
+      if: steps.setup.outputs.publish_to_gar == 'true'
+      uses: google-github-actions/auth@ba79af03959ebeac9769e648f473a284504d9193 # v2.1.10
+      with:
+        workload_identity_provider: ${{ inputs.google_workload_identity_provider }}
+        service_account: ${{ inputs.google_service_account }}
+
+    - name: Configure Docker for Google Artifact Registry
+      if: steps.setup.outputs.publish_to_gar == 'true'
+      shell: bash
+      run: |
+        gcloud auth configure-docker $(echo "${{ inputs.google_ar_image_name }}" | cut -d'/' -f1)
+
+    - name: Build and push
+      id: build-and-push
+      uses: docker/build-push-action@263435318d21b8e681c14492fe198d362a7d2c83 # v6.18.0
+      with:
+        context: ${{ inputs.build_context }}
+        file: ${{ inputs.dockerfile_path }}
+        platforms: linux/${{ inputs.platform }}
+        build-args: ${{ inputs.build_args }}
+        target: ${{ inputs.build_target }}
+        tags: |
+          ${{ steps.setup.outputs.publish_to_ghcr == 'true' && steps.meta-ghcr.outputs.tags || '' }}
+          ${{ steps.setup.outputs.publish_to_gar == 'true' && steps.meta-gar.outputs.tags || '' }}
+        labels: ${{ (steps.setup.outputs.publish_to_ghcr == 'true' && steps.meta-ghcr.outputs.labels) || (steps.setup.outputs.publish_to_gar == 'true' && steps.meta-gar.outputs.labels) || '' }}
+        annotations: ${{ (steps.setup.outputs.publish_to_ghcr == 'true' && steps.meta-ghcr.outputs.annotations) || (steps.setup.outputs.publish_to_gar == 'true' && steps.meta-gar.outputs.annotations) || '' }}
+        cache-from: type=gha,scope=${{ inputs.platform }}
+        cache-to: ${{ steps.setup.outputs.will_publish == 'true' && format('type=gha,mode=max,scope={0}', inputs.platform) || '' }}
+        provenance: ${{ steps.setup.outputs.will_publish == 'true' && 'mode=max' || 'false' }}
+        sbom: ${{ steps.setup.outputs.will_publish == 'true' }}
+        push: ${{ steps.setup.outputs.will_publish == 'true' }}
```

Comment:
I wonder, if the request was a multiplatform build, and they both have `push: true` with the same image name (but with a different `platforms`), wouldn't it override the existing image on the registry?

Reply:
Yep I was right.

<img width="1099" height="107" alt="image" src="https://github.com/user-attachments/assets/25c0917f-890d-4058-a0ea-36e366b4fa71" />

<img width="952" height="513" alt="image" src="https://github.com/user-attachments/assets/9d81f337-07d9-4c99-b24b-012e493d124c" />

👇 The correct one

<img width="865" height="538" alt="image" src="https://github.com/user-attachments/assets/631f8969-3004-47f2-ae41-5f979b8bbcc9" />

Reply:
Darn, I don't see a good approach here since we need arm64 runners to build and push arm64 images and amd64 runners for amd64 platforms. I believe we'll just need to provide guidance that for multi-platform builds a suffix or prefix should be passed in specifying the platform the image is built for to avoid overwriting. Then, we provide a way to combine the images for a multi platform image